src/Razor/test/Microsoft.VisualStudio.LanguageServerClient.Razor.Test/HtmlCSharp/FindAllReferencesHandlerTest.cs

```diff
@@ -0,0 +1,392 @@
+        {
+            // Arrange
+            var called = false;
+            var documentManager = new TestDocumentManager();
+            documentManager.AddDocument(Uri, Mock.Of<LSPDocumentSnapshot>());
+
+            var virtualCSharpUri = new Uri("C:/path/to/file.razor.g.cs");
+            var csharpLocation = GetReferenceItem(100, 100, 100, 100, virtualCSharpUri);
+            var requestInvoker = new Mock<LSPRequestInvoker>(MockBehavior.Strict);
+            requestInvoker
+                .Setup(r => r.ReinvokeRequestOnServerAsync<ReferenceParams, VSReferenceItem[]>(It.IsAny<string>(), It.IsAny<LanguageServerKind>(), It.IsAny<ReferenceParams>(), It.IsAny<CancellationToken>()))
+                .Callback<string, LanguageServerKind, ReferenceParams, CancellationToken>((method, serverKind, definitionParams, ct) =>
+                {
+                    Assert.Equal(Methods.TextDocumentReferencesName, method);
+                    Assert.Equal(LanguageServerKind.CSharp, serverKind);
+                    called = true;
+                })
+                .Returns(Task.FromResult(new[] { csharpLocation }));
+
+            var projectionResult = new ProjectionResult()
+            {
+                LanguageKind = RazorLanguageKind.CSharp,
+            };
+            var projectionProvider = new Mock<LSPProjectionProvider>(MockBehavior.Strict);
+            projectionProvider.Setup(p => p.GetProjectionAsync(It.IsAny<LSPDocumentSnapshot>(), It.IsAny<Position>(), It.IsAny<CancellationToken>())).Returns(Task.FromResult(projectionResult));
+
+            var documentMappingProvider = new Mock<LSPDocumentMappingProvider>(MockBehavior.Strict);
+            documentMappingProvider.Setup(d => d.MapToDocumentRangeAsync(RazorLanguageKind.CSharp, Uri, csharpLocation.Location.Range, It.IsAny<CancellationToken>())).
+                Returns(Task.FromResult<RazorMapToDocumentRangeResponse>(null));
+
+            var referencesHandler = new FindAllReferencesHandler(requestInvoker.Object, documentManager, projectionProvider.Object, documentMappingProvider.Object);
+            var referenceRequest = new ReferenceParams()
+            {
+                TextDocument = new TextDocumentIdentifier() { Uri = Uri },
+                Position = new Position(10, 5)
+            };
+
+            // Act
+            var result = await referencesHandler.HandleRequestAsync(referenceRequest, new ClientCapabilities(), CancellationToken.None).ConfigureAwait(false);
+
+            // Assert
+            Assert.True(called);
+            Assert.Empty(result);
+        }
+
+        private void AssertVSReferenceItem(VSReferenceItem expected, VSReferenceItem actual)
+        {
+            Assert.Equal(expected.Location, actual.Location);
+            Assert.Equal(expected.DisplayPath, actual.DisplayPath);
+        }
+
+        private VSReferenceItem GetReferenceItem(int startLine,
+                                                    int startCharacter,
```

Comment:
indentation

Reply:
Which style rule would this fall under? Ie. what level of indentation is appropriate? Aligning with the first arg?

Reply:
I believe what we follow is do what the editor does when you press enter before the first arg. So that would be something like,
```C#
private VSReferenceItem GetReferenceItem(
    int startLine,
    int startCharacter,
    ...)
```